rules/common_settings.bzl

```diff
@@ -0,0 +1,61 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Common build setting rules that return a BuildSettingInfo with the value of the build setting.
+For label-typed settings, use the native label_flag and label_setting rules."""
+
+BuildSettingInfo = provider(fields = ["value"])
+
+def _impl(ctx):
```

Comment:
Any tests you could write for these, to at least demonstrate their usage?

Reply:
+ common_settings_test which tests flag vs non-flag semantics and the "values" feature